drivers/xen/gntdev-dmabuf.c

```diff
@@ -481,6 +687,111 @@ static int dmabuf_exp_from_refs(struct gntdev_priv *priv, int flags,
 	return ret;
 }
 
+static int dmabuf_release_notifier_cb(struct notifier_block *nfb,
+		unsigned long val, void *gnt_buf)
+{
+	struct gntdev_dmabuf *gntdev_dmabuf = gnt_buf;
+
+	if (!gntdev_dmabuf)
+		return 0;
+
+	dmabuf_map_remove(gntdev_dmabuf->u.map.priv, gntdev_dmabuf);
+	dmabuf_map_wait_obj_signal(gntdev_dmabuf->priv, gntdev_dmabuf);
+	return 0;
+}
+
+static int dmabuf_detach_map(struct gntdev_dmabuf *gntdev_dmabuf)
+{
+	struct dma_buf *dma_buf = gntdev_dmabuf->dmabuf;
+	/* Wait on any implicit fences */
+	if (dma_resv_wait_timeout(dma_buf->resv, true, true,
+						  MAX_SCHEDULE_TIMEOUT) == 0)
+		pr_debug("dma-buf wasn't clear after timeout.Trying to unmap.\n");
```

Comment:
So, this is a normal and expected behaviour? No warning needed?

Reply:
Refactored this part.